File: Java/LeetCode/100/050/item046.java
public class Solution {
    public List<List<Integer>> permute(int[] nums) {
        List<List<Integer>> result = new LinkedList<>();
        permuteCore(nums, result, 0);
        return result;
    }

    private void permuteCore(int nums[], List<List<Integer>> result, int index) {
        if (index == nums.length) {
            List<Integer> list = new ArrayList<>(nums.length);
            for (int n : nums) list.add(n);
            result.add(list);
            return;
        }
        for (int i = index; i < nums.length; i++) {
            swap(nums, index, i);
            permuteCore(nums, result, index + 1);
            swap(nums, index, i);
        }
    }

    private void swap(int nums[], int i, int j) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: C/nowcoder/coding-interviews/item046.cpp
#include <iostream>

using namespace std;

class Solution {
public:
	int LastRemaining_Solution(int n, int m) {
		if (n <= 0 || m <= 0) return -1;
		int result = 0;
		for (int i = 2; i <= n; i++) 
			result = (result + m) % i;
		return result + 1;
	}
};